plugins/source/azure/resources/services/costmanagement/daterange.go

```diff
@@ -0,0 +1,93 @@
+package costmanagement
+
+import (
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/costmanagement/armcostmanagement"
+)
+
+func timePtr(t time.Time) *time.Time {
+	return &t
+}
+
+func ttPtr(v armcostmanagement.TimeframeType) *armcostmanagement.TimeframeType {
+	return &v
+}
+
+func dateRangeToTimeFrame(now time.Time, dr string) (*armcostmanagement.TimeframeType, *armcostmanagement.QueryTimePeriod) {
+	quarterStart := func(t time.Time) time.Time {
+		var m time.Month
+		switch t.Month() {
+		case time.January, time.February, time.March:
+			m = time.January
+		case time.April, time.May, time.June:
+			m = time.April
+		case time.July, time.August, time.September:
+			m = time.July
+		case time.October, time.November, time.December:
+			m = time.October
+		}
+		return time.Date(t.Year(), m, 1, 0, 0, 0, 0, t.Location())
+	}
+
+	now = now.Truncate(24 * time.Hour)
+
+	switch dr {
+	case "Last7Days":
+		now = now.AddDate(0, 0, -1) // ends yesterday
```

Comment:
These dates were verified by the Azure console dates for today, at
<img width="282" alt="Screenshot 2023-03-02 at 09 56 27" src="https://user-images.githubusercontent.com/223029/222395002-6adfa138-24aa-4bff-98b1-64805552de54.png">